kong/plugins/rate-limiting/policies/init.lua

```diff
@@ -84,20 +84,27 @@ return {
         return nil, err
       end
 
-      if conf.redis_password and conf.redis_password ~= "" then
+      local times, err = red:get_reused_times()
+      if err then
+        ngx_log(ngx.ERR, "failed to get connect reused times: ", err)
+        return nil, err
+      end
+      if conf.redis_password and conf.redis_password ~= "" and times == 0 then
         local ok, err = red:auth(conf.redis_password)
         if not ok then
-          ngx_log(ngx.ERR, "failed to connect to Redis: ", err)
+          ngx_log(ngx.ERR, "failed to auth Redis: ", err)
           return nil, err
         end
       end
 
-      if conf.redis_database ~= nil and conf.redis_database > 0 then
-        local ok, err = red:select(conf.redis_database)
-        if not ok then
-          ngx_log(ngx.ERR, "failed to change Redis database: ", err)
-          return nil, err
-        end
+      local db = 0
+      if conf.redis_database ~= nil then
+        db = conf.redis_database
+      end
+      local ok, err = red:select(db)
```

Comment:
The PR makes sense. A comment here to explain _why_ we need to select another database (even after the previous `get_reused_times()` would help. The fact that there is no way to retrieve which database the socket is connected to (from what I can see) is slightly annoying, we are making unnecessary round-trips, so I suspect this to raise eyebrows in the future.